pkg/dumper/postgres/dumper.go

```diff
@@ -49,10 +49,12 @@ func (d *pgDumper) DumpTable(tableName string, rowChan <-chan database.Row) erro
 	insertedRows, err := d.insertIntoTable(txn, tableName, rowChan)
 	if err != nil {
 		err = errors.Wrap(err, "failed to insert rows")
-		if err := txn.Rollback(); err != nil {
-			return errors.Wrap(err, "failed to rollback transaction")
+		if rErr := txn.Rollback(); rErr != nil {
```

Comment:
TBF I was thinking here and indeed it's quite hard to combine errors, I dunno it it's worth, WDYT of this:
```go
        if err != nil {
                 defer func() {
                      if err := txn.Rollback(); err != nil {
			log.WithError(err).Error("failed to rollback")
		     }
                 }()
		return errors.Wrap(err, "failed to insert rows")
	}
```

Reply:
Also if you agree would mind to change the mysql implementation tho?